File: tools/atlas_builder/src/build_target/LibraryBuildTarget.kt
/*
 * Copyright(c) 2023-present, Atlas.
 * Distributed under the MIT License (http://opensource.org/licenses/MIT)
 */

package com.spacelion.atlas_builder.build_target

import java.io.File

class LibraryBuildTarget(name: String,
                         rootDirectory: File,
                         options: Set<String>,
                         oneValueArgs : Map<String, String>,
                         multiValueArgs : Map<String, List<String>>) : BuildTargetBase(name, rootDirectory,
    options, oneValueArgs, multiValueArgs) {

    override fun exportSymbol(): Boolean  = true
}

File: tools/AtlasBuilder/BuildTarget/LibraryBuildTarget.cs
// Copyright(c) 2023-present, Atlas.
// Distributed under the MIT License (http://opensource.org/licenses/MIT)

namespace AtlasBuilder.BuildTarget;

public class LibraryBuildTarget : BuildTargetBase
{
    public LibraryBuildTarget(string targetName, string rootDir) : base(targetName, rootDir)
    {
    }
    
    public LibraryBuildTarget(string targetName, string rootDirectory, HashSet<string> options, Dictionary<string, string> oneValueArgs, Dictionary<string, List<string>> multiValueArgs)
        : base(targetName, rootDirectory, options, oneValueArgs, multiValueArgs)
    {
    }

    public override bool NeedExportSymbol()
    {
        return true;
    }
}